src/main/java/CellContent.java

```diff
@@ -0,0 +1,97 @@
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+
+public class CellContent {
+	public CellType type;
+	public boolean bool;
+	public double numeric;
+	public String string;
+	
+	public boolean wasFormula;
+	
+	public CellContent(Cell cell) {
+		if(cell == null) {
+			type = CellType.BLANK;
+			return;
+		}
+		
+		CellType temp_type = cell.getCellTypeEnum();
+		
+		wasFormula = false;
+		if(temp_type == CellType.FORMULA) {
+			temp_type = cell.getCachedFormulaResultTypeEnum();
+			wasFormula = true;
+		}
+		
+		type = temp_type;
+		switch(type) {
+			case BLANK:
+				break;
+				
+			case BOOLEAN:
+				bool = cell.getBooleanCellValue();
+				break;
+				
+			case NUMERIC:
+				numeric = cell.getNumericCellValue();
+				break;
+				
+			case STRING:
+				string = cell.getStringCellValue();
+				break;
+				
+			default:
```

Comment:
It could be omitted but the Compiler will warn about not having handled all of the Enum values.